scenes/game/local_game.gd

```diff
@@ -4708,6 +4906,163 @@ func handle_strike_effect(card_id :int, effect, performing_player : Player):
+						decision_info.choice.append({
+							"effect_type": "place_buddy_into_space",
+							"buddy_id": performing_player.get_next_free_buddy_id(),
+							"amount": i,
+							"and": and_effect
+						})
+					elif not already_removed_buddy and i in performing_player.buddy_locations:
+						# The player elects to remove this buddy first.
+						decision_info.limitation.append(i)
+						decision_info.choice.append({
+							"effect_type": "remove_buddy",
+							"buddy_id": performing_player.get_buddy_id_at_location(i),
+							"and": {
+								"effect_type": "place_next_buddy",
+								"buddy_name": effect['buddy_name'],
+								"amount": -1, # Additional already in the and effect.
+								"already_removed_buddy": true,
+								"require_unoccupied": require_unoccupied,
+								"destination": destination,
+								"valid_new_positions": valid_new_positions,
+								"and": and_effect
+							}
+						})
+				events += [create_event(Enums.EventType.EventType_ChooseArenaLocationForEffect, performing_player.my_id, 0)]
+			else:
+				# No valid positions to put the buddy, so skip this.
+				# The and effect will occur normally if one exists.
+				pass
+		"move_any_buddy":
+			var move_to_opponent = 'to_opponent' in effect and effect['to_opponent']
+			var move_min = effect['amount_min']
+			var move_max = effect['amount_max']
+			var optional = (move_min == 0)
+			var must_select_other_buddy_first = true
+			change_game_state(Enums.GameState.GameState_PlayerDecision)
+			decision_info.clear()
+			decision_info.type = Enums.DecisionType.DecisionType_ChooseArenaLocationForEffect
+			decision_info.player = performing_player.my_id
+			decision_info.choice_card_id = card_id
+			decision_info.effect_type = "place_next_buddy"
+			decision_info.source = effect['buddy_name']
+			decision_info.choice = []
+			decision_info.limitation = []
+			decision_info.extra_info = must_select_other_buddy_first
+			if optional:
+				decision_info.limitation.append(0)
+				decision_info.choice.append({
+					"effect_type": "pass"
+				})
+			# First, the player has to select a buddy to remove.
+			# Then, they have to place it in a valid space.
+			for location in performing_player.buddy_locations:
+				if location == -1:
+					continue
+				var buddy_id = performing_player.get_buddy_id_at_location(location)
+				var valid_new_positions = []
+				for i in range(MinArenaLocation, MaxArenaLocation + 1):
+					if performing_player.get_buddy_id_at_location(i) != "":
+						# Skip if there is already a buddy here, including self.
+						continue
+					if move_to_opponent and opposing_player.is_in_location(i):
+						valid_new_positions.append(i)
+					else:
+						# Add this space if it is within the amount from the starting buddy location.
+						if abs(location - i) >= move_min and abs(location - i) <= move_max:
+							valid_new_positions.append(i)
+
+				decision_info.limitation.append(location)
+				decision_info.choice.append({
```

Comment:
I assume anything using this effect already checks that you have at least one buddy in play?

Reply:
yes, that is the assumption